src/JustEat.StatsD/SpanStatsDPublisher.cs

```diff
@@ -0,0 +1,196 @@
+using System;
+using System.Text;
+using JustEat.StatsD.EndpointLookups;
+
+namespace JustEat.StatsD
+{
+    public class SpanStatsDPublisher : IStatsDPublisher
+    {
+        private readonly StatsDMessageFormatter _formatter;
+        private readonly IStatsDTransport _transport;
+        private readonly Func<Exception, bool> _onError;
+        private readonly SpanStatsDMessageFormatter _spanFormatter;
+
+        public SpanStatsDPublisher(StatsDConfiguration configuration, IStatsDTransport transport)
+        {
+            if (configuration == null)
+            {
+                throw new ArgumentNullException(nameof(configuration));
+            }
+
+            _transport = transport ?? throw new ArgumentNullException(nameof(transport));
+
+            _formatter = new StatsDMessageFormatter(configuration.Prefix);
+            _spanFormatter = new SpanStatsDMessageFormatter(configuration.Prefix);
+            _onError = configuration.OnError;
+        }
+
+        public SpanStatsDPublisher(StatsDConfiguration configuration)
+        {
+            if (configuration == null)
+            {
+                throw new ArgumentNullException(nameof(configuration));
+            }
+
+            if (string.IsNullOrWhiteSpace(configuration.Host))
+            {
+                throw new ArgumentNullException(nameof(configuration.Host));
+            }
+
+            _formatter = new StatsDMessageFormatter(configuration.Prefix);
+
+            var endpointSource = EndpointParser.MakeEndPointSource(
+                configuration.Host, configuration.Port, configuration.DnsLookupInterval);
+            _transport = new UdpTransport(endpointSource);
+            _onError = configuration.OnError;
+        }
+
+        public void Increment(string bucket)
+        {
+            Span<byte> buffer = stackalloc byte[_spanFormatter.GetMaxSize(bucket)];
```

Comment:
These two lines are repeated again and again in all the methods. Is there a way to use `ref return` or something in a helper method to reduce the copy-paste?